doc/user_guide/tools.rst

```diff
@@ -916,3 +916,33 @@ for example:
       ├── record_by = spectrum
       ├── signal_origin = simulation
       └── signal_type =
+
+
+Speeding up operations
+----------------------
+
+
+Reusing a Signal for output
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Many signal methods create and return a new signal. For fast operations, the
+new signal creation time is non-negligible. However, when the operation is
+repeated many times, for example in a loop, the cumulaive creation time can
+become significant. Therefore, many operations on `Signal` accept an optional
+argument `out`. If an existing signal is passed to `out`, the function output
+will be placed into that signal, instead of being returned in a new signal.
+
+.. code-block:: python
+
+    >>> s = signals.Spectrum(np.arange(10))
+    >>> s_sum = s.sum(0)
+    >>> s_sum.data
+    array(45)
+    >>> s.inav[:5].sum(0, out=s_sum)
+    >>> s_sum.data
+    10
+    >>> s_roi = s.inav[:3]
+    >>> s_roi
+    <Spectrum, title: , dimensions: (|3)>
+    >>> s.__getitem__(slice(None, 5), out=s_roi)
```

Comment:
No such method, should use `inav` or `isig`.

I think it may be also useful to include in the user guide the fact that the `out` signal has to be compatible in terms of number of axes / dimensions. The example works, but it's not obvious that it's not just a convenient way to get a new signal for the purpose of example, but rather the only robust way this syntax should be used.